### DISCUSSION TITLE
[Ring Intercom] Support to pick up and hang up

### DISCUSSION THREAD

**Discussion by roberrML** [tsightler/ring-mqtt#760] [Feature Requests]
Currently, I have two doors connected to the same intercom system. In order to open the second door, I need the ability to hang up after the first ring. This feature is crucial for efficiently managing access to multiple entry points.

Therefore, I kindly request the implementation of a "Pick Up and Hang Up" functionality within the intercom system.

Thanks and best regards

**Comment by tsightler** [comment]
Sorry, but as I don't own an intercom I don't fully understand what this means.  My understanding of "Pick up" and "Hang up" is that it's basically an interactive communications with the door, no different that using the physical handset/speakerphone to speak to the person at the door before buzzing them in.  This project doesn't current support bi-directional communications, even for cameras/doorbells, because the primary use case is for automation, not for interactive use.  

I was not aware that Ring Intercom has any support for mulitiple doors.  Can you please describe the operational steps of mulitple doors via the Ring app in more detail, i.e. what steps do you take to open the first door and what steps do you have to perform differently to open the second door.

**Reply by roberrML** [reply]
My apartment has two doors at the entrance of the building.  One at the entrance to the property, and another in the building itself.  Both doors are opened with the same handset.  The door that I can open is the one that has been called.  The use case would be the following, the person rings on the first door and I open the door using the intercom, however, I have to pick it up and hang up, so that the communication is terminated (or 30s elapse), and this person can call the second door.  That's how it works, unfortunately.  Therefore, I need to be able to hang up, to be able to automate that when I want to open the intercom, I open it and then hang up.  With the Ring app, I can do that, including talking, which is not necessary at all, just being able to open and hang up.  I hope I have explained myself more clearly.
Thanks

**Reply by tsightler** [reply]
OK, I guess it's common for a person to get to the second entrance in less than 30 seconds?

I don't really know how to handle this.  Right now there is no call feature implemented for intercoms in ring-client-api, and you can't hang up if you haven't placed a call.  I have no idea what this looks like from a Ring API perspective and, as neither of us that maintain these projects owns an intercom, the only way to investigate it is if someone would share their intercom with me and be available to test functions at the same time so that I can attempt to reverse engineer what is required for the API by using the Ring app or if someone else does the reverse engineering and provides the API details to us.

My guess, although I don't have any way to know for sure, is that it will use the same streaming API as cameras/doorbells, but just without video, and, since you don't actually need to speak with the person, just perform the action, we could perhaps just start the call and immediately hang up, providing something like an "answer call" switch, which, when toggled on, picks up the call, and when toggled off hangs up.

However, all of this would require quite some effort to work out, and almost certainly will need support in ring-client-api as well.  Unfortunately, lack of time and access to an intercom means this isn't something that is likely to happen any time soon.

**Reply by marcodipasqua** [reply]
Hi @tsightler,
I also would need this feature. It is implemented in Ring App, so I guess an API command is available. I can support this sharing my intercom with you.

**Reply by tsightler** [reply]
Unfortunately this is a non-trivial task and I just don't have time for investigating the API, figuring out how to place calls, adding all of that supporting code to ring-client-api and then coding it up what is required in ring-mqtt as well.  It would also be very difficult to do without having physical access to the intercom device for testing.  Realistically, if intercom users want more capabilities, someone that owns an intercom will likely have to do the work and submit a PR to ring-client-api.  Once it's supported in ring-client-api, then adding support here is more manageable.

If I had more time I'd be happy to investigate, but I just don't right now and I don't see that changing in the next 18-24 months, I barely have time to do minor updates to this project at this point.

**Reply by marcodipasqua** [reply]
Understood, I will try to push this request to ring-client-api.

Thanks,
Marco
